### PR TITLE
[pipeline] add chatglm 

### PR DESCRIPTION
**Pull request by CjhHa1**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

This PR reads chatglm to pipeline

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: tests/kit/model_zoo/transformers/chatglm.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    attention_mask = torch.tensor([[1, 1, 1, 1, 1, 1]])
+    return dict(input_ids=input_ids, attention_mask=attention_mask)
+
+
+# define output transform function
+output_transform_fn = lambda x: x
+
+# define loss function
+loss_fn_for_chatglm_model = lambda x: x.last_hidden_state.sum()
+loss_fn = lambda x: x.logits.sum()
```
Comment (ver217) [inline tests/kit/model_zoo/transformers/chatglm.py:25]: Can you use real loss here?

Comment (CjhHa1) [reply tests/kit/model_zoo/transformers/chatglm.py:25]: in line 24:the real loss is a tensor list
in line 25:as we haven't assign `labels` we can only use logins, which is also a tensor list

File: tests/kit/model_zoo/transformers/chatglm.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+config = ChatGLMConfig(num_layers=1,
+                       padded_vocab_size=65024,
+                       hidden_size=64,
+                       num_attention_heads=8,
+                       rmsnorm=False,
```
Comment (ver217) [inline tests/kit/model_zoo/transformers/chatglm.py:30]: Why disable rmsnorm?

Comment (CjhHa1) [reply tests/kit/model_zoo/transformers/chatglm.py]: fixed

**Review by ver217 — APPROVED** [review]
